### PR TITLE
add lengthFilter

### PR DESCRIPTION
**Pull request by HoloPollock**
add filter for shorts or features set to 80 Min as feature

### CODE REVIEW

**Comment by HoloPollock** [comment]
only backend for now

File: api/index.go [hunk 1]
```diff
@@ -92,28 +97,42 @@ func Handler(w http.ResponseWriter, r *http.Request) {
...
+		scrapefunc = allScrape()
+	}
+	
 	if ignore {
 		if inter {
 			if len(users) == 1 {
-				userFilm, err = scrapeUser(users, false, true)
+					userFilm, err = scrapefunc(users, false, true)
+
+				userFilm, err = scrapefunc(users, false, true)
```
Comment (maxkele) [inline api/index.go:120]: Is this supposed to be done twice?

Comment (HoloPollock) [reply api/index.go:120]: No good catch

**Review by qjack001 — CHANGES_REQUESTED** [review]
use 60min as cutoff

File: api/index.go [hunk 2]
```diff
@@ -35,6 +36,8 @@ const (
 	UNION
 )
 
+const FEATURELENGTH = 80
```
[the remark this answers is not in the document]
Comment (HoloPollock) [reply api/index.go:39]: Cool I agree

File: api/index.go [hunk 3]
```diff
@@ -138,8 +157,27 @@ func Handler(w http.ResponseWriter, r *http.Request) {
...
+	return func(users []string, inter bool, ignore bool) (film, error) {
+		return scrapeUser(users, inter, ignore, false, true)
+	}
+}
+
+func allScrape() func([]string, bool, bool) (film, error){
+	return func(users []string, inter bool, ignore bool) (film, error) {
+		return scrapeUser(users, inter, ignore, false, false)
+	}
+}
```
[the remark this answers is not in the document]
Comment (HoloPollock) [reply api/index.go:175]: Yah this was a hack of using partial funcs to have less gross if statements

**Comment by HoloPollock** [comment]
also if anyone wants to do the front end go ahead

**Review by qjack001 — APPROVED** [review]

**Comment by HoloPollock** [comment]
No one merge yet I wanna do testing cause I change how scraping works

**Comment by HoloPollock** [comment]
the speed slowed down with new changed I want to see if I can fix before merging

**Comment by HoloPollock** [comment]
I cheated and increased the memory and it worked

**Comment by HoloPollock** [comment]
still wanna do some more tests

**Comment by qjack001** [comment]
Could I give you the options in a comma-seperated `ignore` param? Like: `/?users=qjack&ignore=unreleased,shorts,feature&iintersect=true`? Ignore unreleased films if `unreleased` is present, ignore short films if `shorts` is present, ignore feature-length if `feature` is present.

**Comment by HoloPollock** [comment]
> Could I give you the options in a comma-seperated `ignore` param? Like: `/?users=qjack&ignore=unreleased,shorts,feature&iintersect=true`? Ignore unreleased films if `unreleased` is present, ignore short films if `shorts` is present, ignore feature-length if `feature` is present.

Yah that should work I can make that happen

**Comment by qjack001** [comment]
> > Could I give you the options in a comma-seperated `ignore` param? Like: `/?users=qjack&ignore=unreleased,shorts,feature&iintersect=true`? Ignore unreleased films if `unreleased` is present, ignore short films if `shorts` is present, ignore feature-length if `feature` is present.
> 
> Yah that should work I can make that happen

yay

**Comment by HoloPollock** [comment]
so its done and works now I do think we may want to merge this in to the UI branch to test it works with `ignore=unreleased,shorts,feature` it should I tested with postman but still should test integration @qjack001

**Comment by HoloPollock** [comment]
also feel free to look at code for stupid things I did

**Comment by HoloPollock** [comment]
hey so I know why its slower the new page I'm loading is way bigger so it makes it way slower to do may not be worth

**Comment by HoloPollock** [comment]
hey so I know why its slower the new page I'm loading is way bigger so it makes it way slower to do may not be worth

**Comment by HoloPollock** [comment]
hey so I know why its slower the new page I'm loading is way bigger so it makes it way slower to do may not be worth

**Comment by HoloPollock** [comment]
hey so I know why its slower the new page I'm loading is way bigger so it makes it way slower to do may not be worth

**Comment by HoloPollock** [comment]
I dont know another way to get the time though

**Comment by HoloPollock** [comment]
its. difference between loading this [site ](https://letterboxd.com/ajax/poster/film/maggies-plan/menu/linked/125x187/) and the full detail page ex [this](https://letterboxd.com/film/maggies-plan/)

**Comment by HoloPollock** [comment]
lol letterbox please give me API

**Comment by HoloPollock** [comment]
the only thing I can think of to keep this is only load the page when the check boxes are checked

**Comment by qjack001** [comment]
> the only thing I can think of to keep this is only load the page when the check boxes are checked

Only use the heavier page if the `ignore` parameter was given `shorts` or `feature`, yes.

**Comment by qjack001** [comment]
> so its done and works now I do think we may want to merge this in to the UI branch to test it works with `ignore=unreleased,shorts,feature` it should I tested with postman but still should test integration @qjack001 

Once you make the change above, I would just merge into main. It should have no noticeable effect on the site (as it is now). My PR should get the update too, since I believe it uses the prod endpoint.

**Comment by qjack001** [comment]
poke

**Review by maxkele — APPROVED** [review]

**Comment by HoloPollock** [comment]
@qjack001 should be good to merge pls check

**Comment by HoloPollock** [comment]
also closes #69

File: api/index.go [hunk 4]
```diff
@@ -91,30 +104,25 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	_, inter := query["intersect"]
-	_, ignore := query["ignore_unreleased"]
+	ignore, _ := query["ignore"]
+	var ignoreing = toIgnore{}
+
+	if len(ignore) > 0 {  
+		ignoreing = whatToIgnore(ignore[0])
```
Comment (qjack001) [inline api/index.go:111]: lol "ignoreing"

File: api/index.go [hunk 5]
```diff
@@ -253,44 +310,38 @@ func scrape(userName string, ch chan filmSend) {
...
-		name := e.Attr("data-film-name")
-		slug := e.Attr("data-target-link")
+	extensions.RandomUserAgent(ajc)
+	ajc.OnHTML("div#film-page-wrapper", func(e *colly.HTMLElement) {
+		name := e.ChildText("span.frame-title")
+		slug := e.ChildAttr("div.film-poster","data-target-link")
 		img := e.ChildAttr("img", "src")
-		year := e.Attr("data-film-release-year")
+		year := e.ChildAttr("div.film-poster","data-film-release-year")
+		lenght := e.ChildText("p.text-footer")
```
Comment (qjack001) [inline api/index.go:325]: "lenght"

**Review by qjack001 — APPROVED** [review]
great

**Pull request merged**